### ISSUE TITLE
git phlow with jira config is authorizing twice - reproduce and test it

### ISSUE THREAD

**Issue by buep** [Praqma/git-phlow#211]
Feedback from user:

The users .phlow file found in the repo used:

```
[jiraconfig]
    integration_branch = master
    remote = origin
    service = jira
    issue_url = http://jira.customer.com
    delivery_branch_prefix = ready
```

Terminal output:

```
git phlow auth --target jiraconfig
Enter credentials for jira
username: myuser
password:
myuser Successfully authorized: 'git phlow' is now enabled
Enter credentials for jira
username: myuser
password:
responded with 404 Not Found - malformed url
```

Why does it try to authorize twice?

## Deliveries

Two test cases:

* one going through a phlow where the first password is wrong, to shows it asks again or what happens when the password is wrong
* a test case showing authentication is successful and only asks one (happy day scenario)

If any of the above test-cases are not behaving as expect, write a issue with a bug report to fix the code to make the tests green.

**Comment by buep** [comment]
We tried to look into the code @kryptag and I, but we couldn't find anything related to jira in latest sources.
We can see in the code with hit github auth, so unless there are something that isn't renamed/refactored and methods are wrongly named we haven't implemted Jira Auth even though #185 is closed.

We're looking at it now with a debugger, actually just gdb (see https://golang.org/doc/gdb).

Pretty easy, just do

```
gdb go/bin/git-phlow
break Auth.go:30
run
```

but that failed on MAC, @kryptag debugging it. I think we should move this work to a debugging issues to make us able to do debugging.

So when we have the debugger, we will be able to continue with this issues. Estimate seems to be medium if we can debug.

**Comment by kryptag** [comment]
@buep and i found the code that we were looking for, this was because i updated the wrong git-phlow and not the one i the `go` dir

**Comment by buep** [comment]
Indeed, still strange @groenborg did commit #185 without having any jira code.

But back on track to the requested deliveries we want - our test cases :-)

**Comment by kryptag** [comment]
Solved by #220 

This happened because: 
```
if "jira" == strings.ToLower(conf.Service) {
           Auth(INIBlock, plugins.AuthorizeJIRA, plugins.AuthenticateJIRA, 
                "phlow.jirauser","phlow.jiratoken", conf.Service)
}
	 Auth(INIBlock, plugins.AuthorizeGitHub, plugins.AuthenticateGitHub, "phlow.user", 
               "phlow.token", conf.Service)
``` 
Here we check for if its jira and if its jira we auth for jira and then we auth for github after.
All we did to solve this was to add a else if clause for github and an else for a generic msg.
